### ISSUE TITLE
ZC1186 false positive: bare unset is not ambiguous in Zsh

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1372]
ZC1186 ("Use `unset -v` or `unset -f` for explicit unsetting") flags every bare `unset name` on the rationale that it "unsets variables first, then functions". That fall-through is Bash behaviour, not Zsh.

## Zsh semantics

In Zsh, bare `unset name` operates on parameters only. Removing a function requires `unset -f` (or `unfunction`). There is no variable-then-function fall-through, so there is nothing ambiguous to warn about.

Oracle (`zsh 5.9`):

```zsh
foo() { print "function foo" }
foo=barvalue
unset foo
print "var+=${+foo} func+=${+functions[foo]}"
# => var+=0 func+=1   (variable gone, function untouched)
```

`man zshbuiltins`: "unset [ -fmv ] name ... Each named parameter is unset." The `-f` flag is required to reach functions.

## Impact

Bare `unset name` is the ordinary, idiomatic way to unset a variable in Zsh. The rule fires on all of them across the pinned corpora with a rationale that does not hold for the language it targets.

## Fix

Retire the detection. The "ambiguity" it warns about does not exist in Zsh, and `unset -v` is redundant when bare `unset` already targets parameters unambiguously. Keep the kata ID stable and make the check inert, consistent with other rules that imported a Bash-only restriction.
